simphony/io/h5_materials.py

```diff
@@ -0,0 +1,194 @@
+import tables
+import uuid
+
+from simphony.cuds.material import Material
+from simphony.core.data_container import DataContainer
+from simphony.io.data_container_table import DataContainerTable
+from simphony.io.data_container_description import Data
+from simphony.io.data_conversion import convert_from_file_type
+
+MATERIALS_VERSION = 1
+DESCRIPTION_STRING_SIZE = 256
+
+
+class _MaterialDescription(tables.IsDescription):
+    index = tables.StringCol(itemsize=32, pos=0)
+    data = Data()
+    mask = tables.BoolCol(pos=1, shape=(67,))
+    description = tables.StringCol(itemsize=DESCRIPTION_STRING_SIZE)
+
+
+class H5MaterialsItems(DataContainerTable):
+    """ A proxy class to an HDF5 group node with serialised Materials.
+
+    """
+
+    def __init__(self, root):
+        """ Create a proxy object for an HDF5 backed materials table.
+
+        Parameters
+        ----------
+        root : tables.Group
+            The root node where to add materials items and data tables.
+
+        """
+        super(H5MaterialsItems, self).__init__(
+            root, name='materials', record=_MaterialDescription)
+
+    def _retrieve(self, row):
+        """ Return the Material from a table row instance.
+
+        """
+        cuba = self._position_to_cuba
+        mask = row['mask']
+        data = DataContainer({
+            cuba[key]: convert_from_file_type(row['data'][key], cuba[key])
+            for key, valid in enumerate(mask) if valid})
+        uid = uuid.UUID(hex=row['index'], version=4)
```

Comment:
It would be nice to break this into few lines to increase its readability:
masked = {}
for key, valid in enumerate(mask):
    if valid:
        masked[cuba[key]] = convert_from_file_type(row['data'][key], cuba[key])
data = DataContainer(masked)